src/auth/index.js

```diff
@@ -67,7 +67,8 @@ export default class Auth {
   }
 
   refreshToken() {
-    return this.client.doSessionPost().then(json => this.set(json));
+    return Promise.all([this.client.doSessionPost(), this.getLoggedInProviderType()])
```

Comment:
[q] Why does refreshing a token set the provider type?

Reply:
These functions were locked together and would've required a larger refactor.

Reply:
Gotcha 👍